ground/src/main/java/com/google/android/ground/model/geometry/Geometry.kt

```diff
@@ -15,12 +15,9 @@
  */
 package com.google.android.ground.model.geometry
 
-/**
- * Represents types of geometry errors.
- *
- * Typically thrown when a construction does not satisfy definitional constraints for a given geometry.
- */
-sealed class InvalidGeometryException(override val message: String) : Throwable(message)
+import com.google.common.collect.ImmutableList
 
 /** A common ancestor for all geometry types. */
-sealed interface Geometry
\ No newline at end of file
+sealed interface Geometry {
+    abstract val vertices: ImmutableList<Point>
```

Comment:
Do we ever need vertices independent of what type of geometry and structure we're dealing with? I don't think the list of vertices is sufficient for our purposes, except to act as a workaround for legacy code that supported polygon.

Instead of breaking the semantic in callers, I think it would be preferable to have clients use the specific polygon and point coordinates where needed, and in cases where the geometry is a multi-polygon or a polygon with hole, throw a `UnsupportedOperationException` with a TODO(#Issue) to add support. Wdyt?

Reply:
Once we can match on the `geometry` type statically (once all callers are kotlin) we don't, but at the moment, we need to rely on either instanceof checks and casts or go through a unified interface in Geometry. The latter seemed cleaner to me as a temporary state. I'll add a TODO to move toward using specific classes only.

Reply:
Another option would be to move back to subclassing LocationOfInterest, where subclasses have specific geometries.

Reply:
Sgtm, thanks!

Reply:
> Another option would be to move back to subclassing LocationOfInterest, where subclasses have specific geometries.

Unless LOIs need to have different structure depending on the type of geometry they can't contain I can't think of a reason to have the extra complexity. Also, if there are places where views can't use `instanceof` or need a value object (persistence?) we can always add an enum to Geometry to represent the type.